### PR TITLE
Fix play all and shuffle buttons

### LINKED ISSUE #2752: Photo Slideshow no longer working on mobile device
Describe the bug

After recent update to 10.6.4, the photo slideshow was disabled on the android app and android chrome browser.
I noticed a play button is now available on a pc version chrome to start a rapid slideshow, but the button is not found either in the app or the android chrome (even a desktop site is selected). As shown in the follow screenshots:
On PC chrome:
[
PC
](url)
On android chrome:
chrome_android
On android app:
app_android

System (please complete the following information):

Jellyfin host
OS: Ubuntu 20.04 LTS
Jellyfin Version: 10.6.4
Virtualization: Yes

Clients

PC chrome
OS: Windows10 2004 19041.685
Clients: Chrome Browser
Browser: Chrome  87.0.4280.141

Android chrome
OS: Android 10 QKQ1.190828.002
Clients: Chrome Browser
Browser: Chrome  87.0.4280.101

Android app
OS: Android 10 QKQ1.190828.002
Clients: Jellyfin app
App: Version 2.2.3

To Reproduce

Put a photo library on the host
Add the folder to the Jellyfin libraries
Use one of the clients ot open the photo library 
click one of the photos in the library to view it

Expected behavior
For the older version of jellyfin server, the photo slideshow will automaticlly begin on the client side regardless if it's a browser or app on a mobile or PC.
For the Version: 10.6.4, a play button shows up on the bottom of the image to start the slideshow

Additional context
The play/shuffle button on top of the photo library never seem to work, which in my opinion is a real useful feature to have.
button
error

Best regards to all the developpers involved in this project.

### PR DESCRIPTION
**Pull request by thornbill**
**Changes**
* Fixes Play All and Shuffle buttons for Photo libraries
* Removes the buttons from other views that are not supported

**Issues**
Fixes #2752

### CODE REVIEW

File: src/controllers/list.js [hunk 1]
```diff
@@ -791,21 +791,37 @@ class ItemsView {
 
                 const itemType = item ? item.Type : null;
 
-                if (itemType === 'MusicGenre' || params.type !== 'Programs' && itemType !== 'Channel') {
+                if ((itemType === 'MusicGenre' || params.type !== 'Programs' && itemType !== 'Channel')
```
Comment (thornbill) [inline src/controllers/list.js:794]: The list view does not seem to be used for these, but I was afraid to remove this condition. :sweat_smile:

File: src/components/playback/playbackmanager.js [hunk 2]
```diff
@@ -1823,6 +1823,15 @@ class PlaybackManager {
                     SortBy: options.shuffle ? 'Random' : 'SortName',
                     MediaTypes: 'Audio'
                 });
+            } else if (firstItem.IsFolder && firstItem.CollectionType === 'homevideos') {
+                promise = getItemsForPlayback(serverId, mergePlaybackQueries({
+                    ParentId: firstItem.Id,
+                    Filters: 'IsNotFolder',
+                    Recursive: true,
+                    SortBy: options.shuffle ? 'Random' : 'SortName',
+                    MediaTypes: 'Photo,Video',
```
[the remark this answers is not in the document]
Comment (thornbill) [reply src/components/playback/playbackmanager.js]: Good call. I have removed Video items from being included in the query for now.

File: src/controllers/list.js [hunk 3]
```diff
@@ -791,21 +791,37 @@ class ItemsView {
...
+                    // Folder, Playlist views
+                    && itemType !== 'UserView'
+                    // Only Photo (homevideos) CollectionFolders are supported
+                    && !(itemType === 'CollectionFolder' && item?.CollectionType !== 'homevideos')
+                ) {
+                    // Hide Play All buttons
                     hideOrShowAll(view.querySelectorAll('.btnPlay'), false);
```
Comment (dmitrylyzo) [inline src/controllers/list.js:801]: `hideOrShowAll(..., false)` shows the element.
_The name makes it difficult to understand the semantics of the function._

**Review by dmitrylyzo — COMMENTED** [review]
We still have a problem with mixed media, but it is hard to solve right now.
https://github.com/jellyfin/jellyfin-web/blob/8b8cd0f1f9746d9e3c5efe34d5244d53bb69fe3c/src/components/playback/playbackmanager.js#L1815

File: src/controllers/list.js (hunk 3 above)
Comment (thornbill) [reply src/controllers/list.js:801]: I agree the function name is very strange which is why I added comments here. It is probably worthy of a separate PR to rename it.

Comment (thornbill) [reply src/controllers/list.js]: Oh doh. I think I added the comments before I got this working and then didn't update them. Fixed now.

File: src/controllers/list.js [hunk 4]
```diff
@@ -791,21 +791,37 @@ class ItemsView {
 
                 const itemType = item ? item.Type : null;
 
-                if (itemType === 'MusicGenre' || params.type !== 'Programs' && itemType !== 'Channel') {
+                if ((itemType === 'MusicGenre' || params.type !== 'Programs' && itemType !== 'Channel')
+                    // Folder, Playlist views
+                    && itemType !== 'UserView'
+                    // Only Photo (homevideos) CollectionFolders are supported
+                    && !(itemType === 'CollectionFolder' && item?.CollectionType !== 'homevideos')
```
Comment (dmitrylyzo) [inline src/controllers/list.js:798]: Wouldn't that make it easier to perceive?
```suggestion
                    && (itemType !== 'CollectionFolder' || item?.CollectionType === 'homevideos')
```

File: src/controllers/list.js [hunk 5]
```diff
@@ -791,21 +791,37 @@ class ItemsView {
...
+                    // Hide Play All buttons
                     hideOrShowAll(view.querySelectorAll('.btnPlay'), true);
                 }
 
-                if (itemType === 'MusicGenre' || params.type !== 'Programs' && params.type !== 'nextup' && itemType !== 'Channel') {
+                if ((itemType === 'MusicGenre' || params.type !== 'Programs' && params.type !== 'nextup' && itemType !== 'Channel')
+                    // Folder, Playlist views
+                    && itemType !== 'UserView'
+                    // Only Photo (homevideos) CollectionFolders are supported
+                    && !(itemType === 'CollectionFolder' && item?.CollectionType !== 'homevideos')
```
Comment (dmitrylyzo) [inline src/controllers/list.js:811]: Wouldn't that make it easier to perceive?
```suggestion
                    && (itemType !== 'CollectionFolder' || item?.CollectionType === 'homevideos')
```

File: src/controllers/list.js (hunk 4 above)
Comment (thornbill) [reply src/controllers/list.js:798]: It makes it harder to understand for me, but that could just be me. :wink:

**Review by dmitrylyzo — APPROVED** [review]

**Pull request merged**